### DISCUSSION TITLE
an-introduction-to-viewcomponents-a-login-status-view-component/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#185] [Announcements]
# an-introduction-to-viewcomponents-a-login-status-view-component/

In this post I use the example of a Login widget to introduce View Components. I show how to create them, how they're invoked, and the 1.1.0 Tag Helper syntax.

https://andrewlock.net/an-introduction-to-viewcomponents-a-login-status-view-component/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 18 2017, 09:00, in reply to [Sebastiaan Dammann](https://disqus.com/by/Sebazzz/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/185#discussioncomment-6024668)</em>

---

<p>A great point, thanks!:)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 25 2017, 04:00, in reply to [McNulty](https://disqus.com/by/disqus_glfyeQEAkp/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/185#discussioncomment-6024670)</em>

---

<p>Out of interest, how have you seen them abused? </p>

[a comment by an automated account is not shown]

**Comment by Atulin** [comment]
I yearn for something like that, but structured more like Razor Pages. Some `UserHeaderComponent.cshtml` paired with `UserHeaderComponent.cshtml.cs`, without all those folders, duck-typing, all the other song and dance.

**Reply by EmaX093** [reply]
well, i have done that with Hydro

https://github.com/hydrostack/hydro

Check hydro-views / hydro-components, they are identical to razor pages and let you compose components just like that
